yourbench/pipeline/multi_hop_question_generation.py

```diff
@@ -0,0 +1,510 @@
+      2. Optional sampling of multi-hop chunks to manage cost.
+      3. Sending prompt-based requests to a configured LLM for multi-hop Q&A generation.
+      4. Parsing and structuring the results into a new dataset.
+
+    Args:
+        config (Dict[str, Any]):
+            The overall pipeline configuration dictionary, usually loaded
+            from a YAML file. Must include:
+            - pipeline.multi_hop_question_generation.run (bool)
+            - pipeline.multi_hop_question_generation.source_subset (str)
+            - pipeline.multi_hop_question_generation.output_subset (str)
+            - pipeline.multi_hop_question_generation.chunk_sampling (optional dict)
+            - pipeline.multi_hop_question_generation.additional_instructions (str)
+
+    Returns:
+        None. The results are saved as a dataset (e.g., to disk or HF Hub),
+        based on the configuration settings.
+
+    Raises:
+        Exception: Logs errors if something unexpected occurs during
+        inference or dataset processing, but does not halt the entire pipeline.
+    """
+    try:
+        stage_cfg = config.get("pipeline", {}).get("multi_hop_question_generation", {})
+        if not stage_cfg.get("run", False):
+            logger.info("multi_hop_question_generation stage is disabled. Skipping.")
+            return
+
+        # Identify relevant dataset subsets
+        source_dataset_name = smart_get_source_dataset_name("multi_hop_question_generation", config)
+        source_subset = smart_get_source_subset("multi_hop_question_generation", config)
+        output_dataset_name = smart_get_output_dataset_name("multi_hop_question_generation", config)
+        output_subset = smart_get_output_subset("multi_hop_question_generation", config)
+
+        logger.info("Loading dataset for multi-hop QG: '{}'", source_dataset_name)
+        dataset = smart_load_dataset(source_dataset_name, config, source_subset)
+        logger.info("Loaded dataset with {} rows.", len(dataset))
+
+        # Prepare system message for LLM
+        system_msg = {"role": "system", "content": MULTI_HOP_QUESTION_GENERATION_SYSTEM_PROMPT}
+
+        all_inference_calls: List[InferenceCall] = []
+        call_index_map: List[tuple] = []
+
+        def sample_multi_hop_chunks(mh_chunks: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+            """
+            Sample multi-hop chunks from each row based on config settings
+            to control cost (percentage or count-based).
+
+            Args:
+                mh_chunks (List[Dict[str, Any]]):
+                    List of multi-hop chunk dictionaries containing 'chunk_ids'
+                    and 'chunks_text'.
+
+            Returns:
+                List[Dict[str, Any]]:
+                    The potentially sampled subset of multi-hop chunks.
+            """
+            chunk_sampling_cfg = stage_cfg.get("chunk_sampling", {})
+            if not chunk_sampling_cfg:
```

Comment:
If would personally go for "if len(chunk_sampling_cfg) == 0" for the sake of readability